lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -124,3 +133,222 @@ def get(self, request, program_key=None):
         paginated_enrollments = self.paginate_queryset(enrollments)
         serializer = ProgramEnrollmentListSerializer(paginated_enrollments, many=True)
         return self.get_paginated_response(serializer.data)
+
+
+class ProgramSpecificViewMixin(object):
+    """
+    A mixin for views that operate on or within a specific program.
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._program = None
+
+    @property
+    def program(self):
+        """
+        The program specified by the `program_uuid` URL parameter.
+        """
+        if self._program is None:
+            program = get_programs(uuid=self.kwargs['program_uuid'])
+            if program is None:
+                raise Http404()
+            self._program = program
+        return self._program
+
+
+class ProgramCourseRunSpecificViewMixin(ProgramSpecificViewMixin):
+    """
+    A mixin for views that operate on or within a specific course run in a program
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramCourseRunSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._course_run = None
```

Comment:
Again, why do we need these class member `private` variables?

Reply:
we don't! removing